onnx/__init__.py

```diff
@@ -162,17 +164,43 @@ def _get_file_path(f: IO[bytes] | str | os.PathLike) -> str | None:
     return None
 
 
+def _get_serializer(
+    f: str | os.PathLike | IO[bytes] | None, fmt: _SupportedFormat | None
+) -> serialization.ProtoSerializer:
+    """Get the serializer for the given path and format from the serialization registry."""
+    # Use fmt if it is specified
+    if fmt is not None:
+        return serialization.registry.get(fmt)
+
+    if f is None:
```

Comment:
nit: may be `if f is not None` with lines 179-182 would eliminate duplicated line 177?

Reply:
Lines 179-186, I guess

Reply:
Updated logic